ext/pg_text_decoder.c

```diff
@@ -418,4 +608,14 @@ init_pg_text_decoder()
 	pg_define_coder( "Array", pg_text_dec_array, rb_cPG_CompositeDecoder, rb_mPG_TextDecoder );
 	/* dummy = rb_define_class_under( rb_mPG_TextDecoder, "FromBase64", rb_cPG_CompositeDecoder ); */
 	pg_define_coder( "FromBase64", pg_text_dec_from_base64, rb_cPG_CompositeDecoder, rb_mPG_TextDecoder );
+
+#ifdef PG_TEXT_DECODER_TIMESTAMP_EXT
+        /* Document-class: PG::TimestampWithTimeZone < PG::SimpleDecoder */
+        rb_cPG_TimestampWithTimeZone = rb_define_class_under( rb_mPG_TextDecoder, "TimestampWithTimeZone", rb_cPG_SimpleDecoder );
+        rb_define_method( rb_cPG_TimestampWithTimeZone, "decode", pg_text_dec_timestamp_with_time_zone, -1 );
+ 
```

Comment:
Why didn't you use `pg_define_coder()` here?

Reply:
I added a commit to use pg_define_coder: a65a71e40909e6ddb6c4de57369390935066ea79
But I have a question. The decode function does not have the original Ruby string as an argument. In case of a malformed string, I can't return the original string as before. I changed the code to raise an exception as other decode functions do. What do you think ?